drivers: stm32: Convert STM32 drivers to new DT_INST macros

galak opened the pull request:
Convert older DT_INST_ macro use in STM32 drivers to the new
include/devicetree.h DT_INST macro APIs.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
drivers: stm32: Convert STM32 drivers to new DT_INST macros

Convert older DT_INST_ macro use in STM32 drivers to the new
include/devicetree.h DT_INST macro APIs.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

erwango commented on drivers/usb/device/usb_dc_stm32.c:
```diff
@@ -60,34 +62,34 @@ LOG_MODULE_REGISTER(usb_dc_stm32);
+#define DT_USB_CLOCK_BITS		DT_CLOCKS_CELL(DT_INST(0, st_stm32_usb), bits)
+#define DT_USB_CLOCK_BUS		DT_CLOCKS_CELL(DT_INST(0, st_stm32_usb), bus)
+#if DT_NODE_HAS_PROP(DT_INST(0, st_stm32_usb), enable_pin_remap)
+#define DT_USB_ENABLE_PIN_REMAP		DT_PROP(DT_INST(0, st_stm32_usb), enable_pin_remap)
```
Is that working ?
These compats are mutually exclusives, I'm not sure you'll get it work that way.

galak replied:
Why not?  The specific compat (`st_stm32_usb`, `st_stm32_otgfs` or `st_stm32_otghs`) is explicit specified in the macro.

erwango replied:
ok, got it. Will take more time to review this more carefully.

galak replied:
thanks

erwango reviewed (commented):
Quick first feedback

galak committed:
soc: st_stm32: Convert to new DT_INST macros

Convert older DT_INST_ macro use the new include/devicetree.h
DT_INST macro APIs.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

ydamigos reviewed (approved):
Looks good to me
